build_runner/lib/src/asset_graph/serialization.dart

```diff
@@ -238,11 +238,13 @@ class _WrappedAssetNode extends Object with ListMixin implements List {
       case _Field.Id:
         return serializer._assetIdToId[node.id];
       case _Field.Outputs:
-        return node.outputs.map((id) => serializer._assetIdToId[id]).toList();
+        return node.outputs
```

Comment:
In the future eliminating these allocations would be good as well, but not for this pr (for primaryOutputs as well)